sycl/test-e2e/E2EExpr.py

```diff
@@ -0,0 +1,219 @@
+                or (left_unknown and not right)
+                or (not left and right_unknown)
+            )
+
+    def parseAll(self):
+        self.token = next(self.tokens)
+        self.parseOR()
+        self.expect(BooleanExpression.END)
+        return self.findal_unknown_value if self.unknown else self.value
+
+
+import unittest
+
+
+class TestE2EExpr(unittest.TestCase):
+    def test_basic(self):
+        # Non build-only expressions should work the same
+        self.assertTrue(E2EExpr.evaluate("linux", {"linux", "rt_feature"}, False))
+        self.assertTrue(E2EExpr.evaluate("rt_feature", {"linux", "rt_feature"}, False))
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "another_aspect && rt_feature", {"linux", "rt_feature"}, False
+            )
+        )
+        # build-only expressions with no unknowns should work the same
+        self.assertTrue(
+            E2EExpr.evaluate("linux", {"linux"}, True, final_unknown_value=False)
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "linux && windows", {"linux"}, True, final_unknown_value=True
+            )
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "!(windows || zstd)", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        # build-only expressions where unknown affects the resulting value
+        self.assertTrue(
+            E2EExpr.evaluate("rt_feature", {}, True, final_unknown_value=True)
+        )
+        self.assertFalse(
+            E2EExpr.evaluate("rt_feature", {}, True, final_unknown_value=False)
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "rt_feature", {"rt_feature"}, True, final_unknown_value=True
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "rt_feature", {"rt_feature"}, True, final_unknown_value=False
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate("!rt_feature", {}, True, final_unknown_value=False)
+        )
+        self.assertFalse(
+            E2EExpr.evaluate("!!rt_feature", {}, True, final_unknown_value=False)
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "windows || rt_feature", {"linux"}, True, final_unknown_value=True
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "windows || rt_feature", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "linux && rt_feature", {"linux"}, True, final_unknown_value=True
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "linux && rt_feature", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "linux && !(windows || rt_feature)",
```

Comment:
I have hard time mapping that to a possible `REQURIES` directive. I think changing "windows" to something non-OS might help, but even then I'm not sure what would be the best example. Same for the next one.